simplex/docs/reconfiguration.md

```diff
@@ -0,0 +1,497 @@
+
+After replicating the sealing blocks, the onboarding node has the public key(s) of the validator set for each epoch.
+It can thereafter parallelize the replication and finalization certificate verification of all blocks in the chain that reside in between the sealing blocks.
+
+
+## 4. Simplex block encoding
+
+The blocks created by Simplex therefore have the following structure:
+
+
+```
+_________________________
+____________________    |
+|                  |  O |
+|      Inner       |  u |
+|      Block       |  t |
+|                  |  e |
+|__________________|  r |
+|                       |
+|                     B |
+|                     l |
+|  ICM epoch info     o |
+|                     c |
+| Simplex Epoch info  k |
+|                       |
+|   Auxiliary info      |
+|_______________________|
+```
+
+
+```proto
+message SimplexBlock {
+   bytes inner_block = 1; // The inner block built by the VM, opaque to Simplex.
+   OuterBlock outer_block = 2; // The outer block that wraps the inner block without the inner block.
+   bytes protocol_metadata = 3 // The Simplex protocol metadata, set by the Simplex consensus protocol.
+}
+```
+
+where `OuterBlock` is a protobuf message that contains the ICM epoch information, the Simplex epoch information, and auxiliary information:
+
+```proto
+message OuterBlock {
+  ICMEpochInfo icm_epoch_info = 1; // The ICM epoch information.
+  SimplexEpochInfo simplex_epoch_info = 2; // The Simplex epoch information.
+  AuxiliaryInfo auxiliary_info = 3; // The auxiliary information.
+}
+```
+
+The digest of the simplex block is computed as follows:
+
+Let $h_i$ be the hash of the inner block.
+The digest of the Simplex block is the hash of the following encoding:
+
+```proto
+message HashPreImage {
+   bytes h_i = 1; // The inner block hash
+   OuterBlock outer_block = 2;
+   bytes protocol_metadata = 3;
+}
+```
+
+
+This way of hashing the block allows any holder of a finalization certificate for the block to authenticate the block while hiding the content of the inner block.
+
+The `ICM epoch info` is encoded as a canoto message with the following schema:
+
+```proto
+message ICMEpochInfo {
+   uint64 epoch_start_time = 1; // The start time
+   uint64 epoch_number = 2; // The epoch number
+   uint64 p_chain_epoch_height = 3; // The P-chain height of the epoch
+```
+
+The Simplex epoch information is a canoto encoded message with the following schema:
+
+```proto
+message ObservedPChainHeight {
+   uint32 node_index = 1; // The index of the node that proposed the block
+   uint64 p_chain_height = 2; // The P-chain height sampled by the node
+}
+
+message NodeBLSMapping {
+    bytes node_id = 1; // The nodeID
+    bytes bls_key = 2; // The BLS key of the node
+  }
+
+message BlockValidationDescriptor {
+  oneof BlockValidationType {
+      repeated NodeBLSMapping aggregated_membership = 1; // The BLS keys of the nodes in the next epoch
```

Comment:
It isn't valid to include a `repeated` field in a `OneOf`

Reply:
thanks, fixed